test/test.bash

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+DIR="$(cd "$(dirname "${0}")/.." && pwd)"
+cd "${DIR}"
+
+mkdir -p "${DIR}/tmp/test/bin"
+trap 'rm -rf "${DIR}/tmp/test"' EXIT
+
+PATH="${DIR}/tmp/test/bin:$PATH"
+
+cp test/buf.bash tmp/test/bin/buf
+chmod +x tmp/test/bin/buf
+
+# prevent the GITHUB_SHA set by actions from being used in test
+unset GITHUB_SHA
+
+test_push() {
```

Comment:
This is neat - do we need a new style guide on testing bash?

Reply:
Maybe, but I hope to keep the amount of bash we maintain small enough that it isn't worth having a style guide.

Reply:
I'm mostly kidding - I just think this is cool :)